packages/vit-s/src/hooks.js

```diff
@@ -322,3 +322,58 @@ export function useGetObsInfo(obsType, obsLabelsTypes, obsLabelsData, obsSetsMem
     return null;
   }, [obsType, obsLabelsTypes, obsLabelsData, obsSetsMembership]);
 }
+
+
+export const useMappedGeneList = (geneObject, featureMap) => {
+  const [mappedGeneList, setMappedGeneList] = useState(geneObject);
+  useEffect(() => {
+    const processFeatureLabelMap = async () => {
+      // Assuming featureMap is set for symbolic genes
+      if (featureMap !== undefined) {
+        setMappedGeneList(geneObject);
+        return;
+      }
+      try {
+        let ensbGenes = false;
+        if (Array.isArray(geneObject)) ensbGenes = geneObject?.some(value => value?.toUpperCase().startsWith('ENSG'));
+        else {
+          ensbGenes = Object.entries(geneObject)?.some(([key, value]) => key.toLowerCase().includes('gene') && value?.toUpperCase().startsWith('ENSG'));
+        }
+        // console.log(ensbGenes)
+        if (!ensbGenes) return;
+
+        let updatedGeneObject = { ...geneObject };
+        const response = await fetch('https://vitessce-resources.s3.us-east-2.amazonaws.com/genes_filtered.json');
```

Comment:
I think it would be valuable to extract the fetching aspect of this functionality out into a separate effect/hook; this would allow us to simplify some of the conditional logic in this effect, make it easier to follow, and more closely follow single-responsibility principles.

e.g. something like:

```
const [remoteGeneList, setRemoteGeneList] = useState();
useEffect(() => {
	const fetchFeatureLabelMap = async () => {
		/* logic for fetching, trimming, and filtering the results here, ending with... */
		setRemoteGeneList(jsonData)
	}
	fetchFeatureLabelMap();
}, []); // this approach would run once per reference to this hook, so maybe this could be part of a higher level context to avoid multiple fetches
```

Another idea practice to consider is to attach an AbortController to the fetch call so that it correctly aborts the first request that's sent in strict mode; remote state fetching libraries like react-query/swr do this for us under the hood, but it needs to be done manually if using a `useEffect` fetch. https://medium.com/@icjoseph/using-react-to-understand-abort-controllers-eb10654485df

Reply:
This one is a little tricky. So the fetching is needed only when there are ensemble-gene-ids. One way to handle this would be to check the data for ensemble-ids before hand and create a context with the fetched genes and use that (the file is 1.5MB), not sure if we want to keep it in the context and if it's overkill?